lnwallet/channel.go

```diff
@@ -934,13 +953,6 @@ func (lc *LightningChannel) diskCommitToMemCommit(isLocal bool,
 		commit.dustLimit = lc.channelState.RemoteChanCfg.DustLimit
 	}
 
-	// Finally, we'll re-populate the HTLC index for this state so we can
-	// properly locate each HTLC within the commitment transaction.
-	err = commit.populateHtlcIndexes(lc.channelState.ChanType)
```

Comment:
Why was this removed?

Reply:
We still needs this when we go to write a commitment to disk, as we need the `outputIndex` value.

Reply:
Or is this this is unnecessary, since we have already signed this commitment, and won't need to write it to disk _again_?

Reply:
> since we have already signed this commitment, and won't need to write it to disk again?

correct, we've already written the diff at this point, plus we don't have the information available to repopulate them properly w/o access to the sorted cltvs anyway.

however, this doesn't repopulate the `outgoingHTLCIndex` and `incomingHTLCIndex` maps, perhaps those _should_ be repopulated in case we don't receive the remote party's signatures before the link is restarted?

Reply:
test case with restart added, doesn't appear that we need to repopulate